cocotb/triggers.py

```diff
@@ -631,16 +615,13 @@ class Waitable(object):
     supporting Python versions (>=3.3).
```

Comment:
You can remove this comment in the docstring since cocotb supports 3.5+ now.

Reply:
If we're ok with scope creep, we can also add `collections.abc.Awaitable` as a base class.

Reply:
Do we really need Waitable anymore? They could easily be rewritten as coroutines and we could remove that special case from the scheduler.

Reply:
The purpose of `Waitable` is to be reusable coroutines - you add them arguments just once, and to can then await them repeatedly.

Essentially, the idea is they behave like triggers, but we get to implement them using coroutines.

Reply:
Added in `Awaitable` base classes to `Trigger` and `Waitable`.